### PR TITLE
added passthru option to read function

### PR DESCRIPTION
**Pull request by zolex**
This small feature adds the ability to output the response from the read method "in realtime" which is good for commands which take a while and send the output while running.

### CODE REVIEW

File: phpseclib/Net/SSH2.php [hunk 1]
```diff
@@ -2664,7 +2664,7 @@ function _get_open_channel()
      * @throws \RuntimeException on connection error
      * @access public
      */
-    function read($expect = '', $mode = self::READ_SIMPLE)
+    function read($expect = '', $mode = self::READ_SIMPLE, $passthru = false)
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: phpdoc needed

File: phpseclib/Net/SSH2.php [hunk 2]
```diff
@@ -2694,6 +2694,12 @@ function read($expect = '', $mode = self::READ_SIMPLE)
                 $this->in_request_pty_exec = false;
                 return $response ? $this->_string_shift($this->interactiveBuffer, strlen($this->interactiveBuffer)) : false;
             }
+            
+            if ($passthru === true) {
+
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: cs

File: phpseclib/Net/SSH2.php (hunk 1 above)
Comment (zolex) [inline phpseclib/Net/SSH2.php]: added

File: phpseclib/Net/SSH2.php (hunk 2 above)
Comment (zolex) [inline phpseclib/Net/SSH2.php]: hm?

Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: no random extra line

**Comment by GrahamCampbell** [comment]
Don't change the file permissions.

File: phpseclib/Net/SSH2.php (hunk 2 above)
Comment (zolex) [inline phpseclib/Net/SSH2.php]: removed

**Comment by zolex** [comment]
file permissions reverted

**Comment by terrafrost** [comment]
Why not just do `if ($passthru)`? Doing `if ($passthru === true)` will make it so $passthru can't be 1. That's not a huge issue, I suppose, but is there an overwhelming reason to require that it be bool(true)?

Like `sha1`'s second parameter is a boolean but `sha1('', true)` and `sha1('', 1)` output the same thing.  With `sha1` the second parameter will be treated as `true` so long as it's castable to `true`. But with this PR we're not going to be doing that?

Also, this should be against the 1.0 branch. There's no reason this feature can't exist in that branch. It's not a BC break or anything.

**Pull request closed without merging**